WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -2525,18 +2494,105 @@ local function flattenRegionOptions(allOptions, isGroupTab)
   return result;
 end
 
-local function removePrefix(input)
+local function parsePrefix(input, data, create)
+  local subRegionIndex, property = string.match(input, "^sub%.(%d+)%..+%.(.+)")
+  subRegionIndex = tonumber(subRegionIndex)
+  if subRegionIndex then
+    if create then
+      data.subRegions = data.subRegions or {}
+      data.subRegions[subRegionIndex] = data.subRegions[subRegionIndex] or {}
+    else
+      if not data.subRegions or not data.subRegions[subRegionIndex] then
+        return nil
+      end
+    end
+    return data.subRegions[subRegionIndex], property
+  end
   local index = string.find(input, ".", 1, true);
   if (index) then
-    return string.sub(input, index + 1);
+    return data, string.sub(input, index + 1);
+  end
+  return data, input
+end
+
+local function AddSubRegionImpl(data, subRegionName)
+  data.subRegions = data.subRegions or {}
+  if WeakAuras.subRegionOptions[subRegionName] and WeakAuras.subRegionOptions[subRegionName] then
+    if WeakAuras.subRegionOptions[subRegionName].supports(data.regionType) then
+      local default = WeakAuras.subRegionTypes[subRegionName].default
+      local subRegionData = type(default) == "function" and default(data.regionType) or CopyTable(default)
+      subRegionData.type = subRegionName
+      tinsert(data.subRegions, subRegionData)
+      WeakAuras.Add(data)
+      WeakAuras.ReloadOptions2(data.id, data)
+    end
+  end
+end
+
+local function AddSubRegion(data, subRegionName)
+  WeakAuras.ApplyToDataOrChildData(data, AddSubRegionImpl, subRegionName)
+  WeakAuras.ReloadOptions2(data.id, data)
+end
+
+local function AddOptionsForSupportedSubRegion(regionOption, data, supported)
+  local result = {}
+  local order = 1
+  result.__order = 300
+  result.__title = L["Add Sub Regions"]
+  for subRegionType in pairs(supported) do
```

Comment:
This adds a button per potential sub region type. That'll look horrible with more than 2 potential sub region types, but we can figure that out once there's more than 1 :)